### PR TITLE
Rust 1.75.0

### PR DESCRIPTION
**Pull request by salaheldinsoliman**
Update to Rust 1.75.0

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
       uses: actions/checkout@v4
       with:
         submodules: recursive
-    - uses: dtolnay/rust-toolchain@1.74.0
+    - uses: dtolnay/rust-toolchain@1.82.0
```
Comment (seanyoung) [inline .github/workflows/release.yml:18]: 1.75 should suffice here and everywhere below (keep the 1.82.0 fixes though).

Problem is that you are forcing anyone who uses the solang or solang-parser crate to update to 1.82.0, when it should work with older versions.

Comment (salaheldinsoliman) [reply .github/workflows/release.yml:18]: Solang depends on a crate which requires `1.75.0`: https://github.com/hyperledger-solang/solang/actions/runs/11792189166/job/32845426551?pr=1664#step:6:505

Should I upgrade to `1.75.0` instead?

Comment (seanyoung) [reply .github/workflows/release.yml:18]: so it should be the lowest version that works

**Review by seanyoung — APPROVED** [review]

**Comment by seanyoung** [comment]
Looks good, let's merge :)

**Pull request merged**